src/ClientOptionsTrait.php

```diff
@@ -124,9 +143,19 @@ private function buildClientOptions(array $options)
         if (isset($options['transportConfig']['grpc'])) {
             $options['transportConfig']['grpc'] += $defaultOptions['transportConfig']['grpc'];
             $options['transportConfig']['grpc']['stubOpts'] += $defaultOptions['transportConfig']['grpc']['stubOpts'];
+            $options['transportConfig']['grpc']['logger'] = null;
+            
+            if (isset($options['logger'])) {
+                $options['transportConfig']['grpc']['logger'] = $options['logger'];
+            }
```

Comment:
One line instead of 4 is a win.
```suggestion
            $options['transportConfig']['grpc']['logger'] = $options['logger'] ?? null;
```